tools/sdk/ld/eagle.app.v6.common.ld.h

```diff
@@ -162,6 +162,24 @@ SECTIONS
     *(.rodata._ZTV*) /* C++ vtables */
 #endif
 
+    . = ALIGN(4);       /* this table MUST be 4-byte aligned */
+    /*  C++ constructor and destructor tables, properly ordered:  */
+    __init_array_start = ABSOLUTE(.);
+    KEEP (*crtbegin.o(.ctors))
+    KEEP (*(EXCLUDE_FILE (*crtend.o) .ctors))
+    KEEP (*(SORT(.ctors.*)))
+    KEEP (*(.ctors))
+    __init_array_end = ABSOLUTE(.);
+    KEEP (*crtbegin.o(.dtors))
+    KEEP (*(EXCLUDE_FILE (*crtend.o) .dtors))
+    KEEP (*(SORT(.dtors.*)))
+    KEEP (*(.dtors))
+    . = ALIGN(4);       /* this table MUST be 4-byte aligned */
+    _bss_table_start = ABSOLUTE(.);
+    LONG(_bss_start)
+    LONG(_bss_end)
+    _bss_table_end = ABSOLUTE(.);
+
```

Comment:
Was this something from old SDK? Don't see anything referring to this

Reply:
SDK 2.2.1 will zero that range during `user_start()`. I think the globals without values are grouped there.

Reply:
This range is zeroed 1 byte at a time, so flash storage will not work.

Reply:
I mean `_bss_table_start` and `end`. It injects addresses of bss\_start and end between the two, but those are still in RAM
(see LONG(addr) -> https://sourceware.org/binutils/docs/ld.html#Output-Section-Data)